docker-compose-rule-core/src/test/java/com/palantir/docker/compose/execution/DockerComposeShould.java

```diff
@@ -166,37 +172,69 @@ public void throw_exception_when_down_fails_for_a_reason_other_than_the_command_
     @Test
     public void use_the_remove_volumes_flag_when_down_exists() throws IOException, InterruptedException {
         compose.down();
-        verify(executor).execute("down", "--volumes");
+        verify(dockerComposeExecutor).execute("down", "--volumes");
     }
 
     @Test
-    public void parse_the_ps_output_on_ports() throws IOException, InterruptedException {
+    public void get_correct_ports() throws IOException, InterruptedException {
```

Comment:
Not sure about the value of this test - it mocks out a lot of behaviour and I wonder if an integeration test against docker or just testing the parsing directly would be more valuable.

Reply:
I could trade this one out for an integration test, it would be easy enough to create a `docker-compose.yml` with various port setups and then check all the parsing without mocking. Will add that